Rich-Text-Editor-(RTE)-Overview.md

```diff
@@ -1,117 +1,137 @@
-This function is executed while initialising a widget, after a widget instance is created, but before it is ready. It is executed before the first time when data function is exceuted. It reads values from component attributes and saves them.
-
-# Hidden components
+`CkEditor4RteComponent` is the Angular component, defined in [`ck-editor-4-rte.component.ts`](https://github.com/oppia/oppia/blob/develop/core/templates/components/ck-editor-helpers/ck-editor-4-rte.component.ts). There we define a `ckConfig` object that configures the editor. This configuration includes the `toolbar` attribute, which specifies the order in which rich text components appear in the editor toolbar.
+
+Near the end of the `ngAfterViewInit()` function, we use `ck.on` to specify the following callback functions:
+
+* The function passed to `ck.on('instanceReady', ...)` is executed when any new instance of CKEditor is ready. For example it will be executed when we click on the content of a card, for each option that we have for multiple choice options etc. Here we can add the css and icons to the buttons in the toolbar.
+* The function passed to `ck.on('change', ...)` is executed whenever the content of CKEditor changes. It handles user updates to the rich text.
+
+### Adding a new plugin
+
+To add a new plugin, follow these steps:
+
+1. Add the plugin to the `manifest.json` to download it. `SharedSpace` can be looked as an example. Alternatively, define your own custom plugin like we did with `pre`.
+2. Add a call to `CKEditor.plugins.addExternal` to the component's `ngAfterViewInit()` method.
+3. Add this plugin to `extraPlugins` in `ckConfig`.
+
+#### Adding rich text components to CKEditor
+
+Rich text components are added to CKEditor in [`ck-editor-4-widgets.initializer.ts`](https://github.com/oppia/oppia/blob/develop/core/templates/components/ck-editor-helpers/ck-editor-4-widgets.initializer.ts). The components are dynamically added to CKEditor as [widgets](https://docs.ckeditor.com/ckeditor4/latest/guide/widget_sdk_intro.html). We use the `getRichTextComponents()` function to obtain the information of each rich text component and construct there respective widgets. We also have a function `isInlineComponent()` to check whether a rich text component is inline or block component.
+
+`componentTemplate` defines a template to wrap the rich text components. Inline components are wrapped in a span, and block components are wrapped in a div.
+
+The plugins are added to CKEditor in the line `CKEDITOR.plugins.add(ckName, {`. The `init` function passed to `add()` is executed when the plugin is initialised and it adds a widget for each component using the line `editor.widgets.add(ckName, {`. `editor.widgits.add` is passed an object with the following attributes:
+
+* `button`: The button label for the widget.
+* `inline`: Whether the widget is an inline component or block component. The link and the math component are inline and the rest are declared block components.
```

Comment:
What is the significance of the word "declared" here?

Reply:
It's not useful. I removed it